### PR TITLE
DOI minting

### LINKED ISSUE #877: DOI minting for works

### PR DESCRIPTION
**Pull request by jamesvanmil**
Closes #877 

Ports work from Curate to implement DOI minting.

* Major model additions implemented via the `concerns` pattern, to ease the addition to future works.
* Major model tests are implemented via `shared examples`, to ease addition of multiple work types.

### CODE REVIEW

**Comment by jamesvanmil** [comment]
A comment on my test coverage decrease: https://coveralls.io/builds/8874970/source?filename=app%2Factors%2Fcuration_concerns%2Factors%2Fwork_actor.rb

I struggled for a while with unit testing this method, but to do it I would need to spend a significant amount of time cracking open the Actor Stack. I decided instead to make this method private (since it's not called from outside the actors and to cover this with feature tests.

**Comment by jamesvanmil** [comment]
To Do: Make Publisher required when selecting DOI mining. Should not be able to submit unless it's filled.

File: app/views/curation_concerns/works/_work.html.erb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 <%# This is a search result view %>
 <%= render 'catalog/document', document: work, document_counter: work_counter  %>
+
```
Comment (jamesvanmil) [inline app/views/curation_concerns/works/_work.html.erb]: Remove me.

**Comment by crowesn** [comment]
Is this just my local? 

```
  Sufia::PermissionTemplate Load (0.3ms)  SELECT  "permission_templates".* FROM "permission_templates" WHERE "permission_templates"."admin_set_id" = ? LIMIT 1  [["admin_set_id","
admin_set/default"]]
Completed 404 Not Found in 194ms (ActiveRecord: 0.8ms)

ActiveRecord::RecordNotFound (Couldn't find Sufia::PermissionTemplate):
  app/controllers/concerns/scholar/works_controller_behavior.rb:13:in `create'
```

**Pull request merged**